kernel/src/drivers/terminal/mod.rs

```diff
@@ -0,0 +1,316 @@
+use core::ops::Add;
+use core::fmt;
+use core::result::Result;
+use spin::Mutex;
+use drivers::vga::{self, VgaWriter};
+use color::Color;
+
+pub static WRITER: Mutex<VgaWriter> = Mutex::new(VgaWriter::new());
```

Comment:
Does this *need* to be public? How about just giving it to `stdout` directly?

Reply:
It needs to be public - any terminal you create that writes to VGA would need this.

Reply:
OK, then this should be in the vga module.